### DISCUSSION TITLE
Is there a way to exit a program with a non-zero return code?

### DISCUSSION THREAD

**Discussion by mortax** [modular/modular#1136] [General]
I can't find anything in Mojo equivalent to:

```
import sys

sys.exit(13)
```

Am I just missing it?

**Comment by rarebreed** [comment]
So, until mojo makes this more friendly,  you will need to make a System Call to do this.  

```python
# nonzero.mojo
from sys.intrinsics import external_call

fn main() raises:
    external_call["exit", Int, Int](20)
```

```bash
~/Projects/anting via 🐍 v3.10.13 (.venv) 
❯ : mojo build nonzero.mojo
/home/stoner/Projects/anting/nonzero.mojo:4:36: warning: 'Int' value is unused
    external_call["exit", Int, Int](20)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~
    _ = 
```

Then you can run it

```bash
➜  ./nonzero
➜  echo $?  
20
```

Took me awhile to figure out that mojo requires `def main()` or `fn main()` to return None.  The `external_call` is mojo's way to let you make a libc call under the hood.

**Comment by mortax** [comment]
ah ha. nice. i tried doing it through Python but that didn't actually do anything do to the way it's wrapped (i presume).

i _knew_ there must be some (possibly non-beautiful) way to do it. that worked perfectly, thanks!
